execution/state.go

```diff
@@ -172,6 +181,84 @@ func MakeGenesisState(db dbm.DB, genesisDoc *genesis.GenesisDoc) (*State, error)
 	return s, nil
 }
 
+func (s *State) LoadDump(filename string, genesisDoc *genesis.GenesisDoc) error {
+	f, err := os.OpenFile(filename, os.O_RDONLY, 0644)
+	if err != nil {
+		return err
+	}
+
+	tx := exec.TxExecution{
+		TxType: payload.TypeCall,
+		TxHash: make([]byte, 32),
+	}
+
+	for {
+		var row dump.Dump
+
+		_, err = cdc.UnmarshalBinaryLengthPrefixedReader(f, &row, 0)
+		if err != nil {
+			break
+		}
+
+		if row.Account != nil {
+			if row.Account.Address == crypto.ZeroAddress {
+				if genesisDoc != nil {
+					genesisDoc.GlobalPermissions.Base = row.Account.Permissions.Base
+				}
+				continue
+			}
+
+			s.writeState.UpdateAccount(row.Account)
+
+			if genesisDoc != nil {
+				found := false
+
+				if len(row.Account.PublicKey.PublicKey) == 0 {
+					// no public key, so not an account
```

Comment:
hmm... Can we say 'so not a known account' - we lazily harvest pub keys as we see them, we can't say it's not an account (even if we are clear on what exactly that means)

Reply:
Removed